### PR TITLE
Constify add / sub core

### PR DESCRIPTION
**Pull request by kaidokert**
#58

## Summary by Sourcery

Make FixedUInt core add/sub operations and panic helper usable in const contexts and operate directly on underlying word arrays.

New Features:
- Expose const-compatible add and sub helper functions over raw machine-word arrays.
- Allow maybe_panic to be invoked from const-evaluated code paths.
- Add const wrappers for FixedUInt overflowing add/sub and Add/Sub operators for use in const contexts.

Enhancements:
- Make FixedUInt internal word array field accessible within the crate for reuse by shared helpers.
- Reuse shared const add/sub implementations across trait impls instead of per-method logic.

Tests:
- Add unit tests validating const add/sub array helpers for multiple word sizes and overflow/underflow behavior.
- Add unit tests for const OverflowingAdd/OverflowingSub implementations and Add/Sub operators, including nightly-only compile-time const evaluations.

## Summary by CodeRabbit

* **New Features**
  * Addition and subtraction for fixed-size integers are now available in compile-time (const) contexts and via standard arithmetic operators.

* **Behavior Changes**
  * Const-aware arithmetic will panic on overflow in const contexts, matching runtime behavior.
  * Internal visibility was adjusted at the module boundary to enable const implementations.

* **Tests**
  * Added compile-time tests covering no-overflow, carry/borrow propagation, and overflow scenarios.

### CODE REVIEW

File: src/fixeduint/add_sub_impl.rs [hunk 1]
```diff
@@ -242,4 +294,105 @@ mod tests {
...
+        let (result, overflow) = const_overflowing_add(&a, &b);
+        assert_eq!(result, FixedUInt::<u8, 2>::from(15u8));
+        assert!(!overflow);
+
+        // With overflow
+        let a = FixedUInt::<u8, 2>::from(255u8);
+        let b = FixedUInt::<u8, 2>::from(255u8);
+        let a = a + FixedUInt::<u8, 2>::from(1u8); // 256
+        let (result, overflow) = const_overflowing_add(&a, &b);
+        assert!(overflow || result > FixedUInt::<u8, 2>::from(0u8)); // Either overflows or wraps
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/add_sub_impl.rs]: done
